.github/workflows/tests-smoke.yaml

```diff
@@ -129,6 +129,41 @@ jobs:
             until docker manifest inspect quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/$image:${{ steps.sha.outputs.sha }} &> /dev/null; do sleep 45s; done
           done
 
+      - name: Generate CA, Server and Client certs for Operator Prometheus
```

Comment:
I wonder if we could have used `certgen` for this (and also to offer it as an option to the user)? We have lots of certificate generating infrastructure in Cilium already. But this PR is already big enough as is, I think it's fine to move that to potential future work.